libkbfs/folder_branch_ops.go

```diff
@@ -1631,24 +1625,58 @@ func (fbo *folderBranchOps) readyBlockMultiple(ctx context.Context,
 
 func (fbo *folderBranchOps) unembedBlockChanges(
 	ctx context.Context, bps *blockPutState, md *RootMetadata,
-	changes *BlockChanges, uid keybase1.UID) (err error) {
+	changes *BlockChanges, uid keybase1.UID) error {
 	buf, err := fbo.config.Codec().Encode(changes)
 	if err != nil {
-		return
+		return err
 	}
-	block := NewFileBlock().(*FileBlock)
-	block.Contents = buf
-	info, _, err := fbo.readyBlockMultiple(
-		ctx, md.ReadOnly(), block, uid, bps)
-	if err != nil {
-		return
+
```

Comment:
There are now two cases here: the changes fit into a single block, or they need multiple blocks. It seems that the code for these two cases are intermingled, though, which makes it confusing. Perhaps something like this would be clearer:

```
copied := fb.config.BlockSplitter.CopyUntilSplit(block, false, buf, 0)
if copied == len(buf) {
  // Handle fit into single block case.
  return bps, nil
}

// Handle multiple block case, which is similar to the existing code, but with some checks not needed.
```

The only cost is having multiple calls to `CopyUntilSplit`, but that seems fine to me. What do you think?